base_geoengine/static/src/js/views/geoengine/layers_panel/layers_panel.esm.js

```diff
@@ -30,15 +30,7 @@ export class LayersPanel extends Component {
          * in the database and add them to the store.
          */
         onWillStart(async () => {
-            this.isGeoengineAdmin = await this.user.hasGroup(
-                "base_geoengine.group_geoengine_admin"
-            );
-            const result = await this.orm.call(
-                this.props.model,
-                "get_geoengine_layers",
-                []
-            );
-            this.state.geoengineLayers = result;
+            await Promise.all([this.load()]);
```

Comment:
@kouffsamuel IMO your `Promise.all` is not well written. To allows // calls of the `user.hasGroup` and `orm.call` methods this should be written as:

```javascript

    await Promise.all([this.loadIsAdmin(), this.loadLayers() ]);
  });
}

async loadIsAdmin() {
  return this.user.hasGroup(
      "base_geoengine.group_geoengine_admin"
  ).then(function(result){
      this.isGeoengineAdmin = await 
  });
}

async loadLayers () {
  return this.orm.call(
      this.props.model,
      "get_geoengine_layers",
      []
  ).then(function(result){
    this.state.geoengineLayers = result;
});
}